.github/workflows/test.yml

```diff
@@ -93,3 +93,37 @@ jobs:
         echo
       env:
         COMBINATION: 1
+
+  install-dir-test:
+    runs-on: windows-latest
+    name: 'Check install directory behaviour'
+
+    strategy:
+      matrix:
+        include:
+          - install-dir: C:\tools\cygwin
+          - install-dir: D:\cygwin64
+
+    steps:
+    - run: git config --global core.autocrlf input
+      shell: pwsh -command ". '{0}'".
+
+    - uses: actions/checkout@v2
+
+    - name: Install Cygwin
+      uses: ./
+      with:
+        install-dir: "${{ matrix.install-dir }}"
+
+    - name: Check working directory
+      run: |
+        if [[ "$(cygpath -aw /)" == '${{ matrix.install-dir }}' ]]; then
+          echo "Installed in $(cygpath -aw /)"
+        else
+          exit 1
+        fi
+      # Only specify the shell by name; GitHub actions should find the correct
+      # executable thanks to the action setting the path appropriately.
```

Comment:
That's rather the point of this part of the test! There definitely is a bash in the path already (I suspect from Git for Windows, but I've not actually checked), but the setting `env:GITHUB_PATH` in the action script means that Cygwin's bash should be the first one found, and if it isn't, that'd indicate something had gone wrong with the paths.

Reply:
Ah, makes sense.

I don't think I'd fully absorbed that GITHUB_PATH is prepended to the PATH.  

I've tweaked the doc a bit since writing `bash` is clearly preferable to `C:\cygwin\bin\bash.exe`, and made this test check OSTYPE as well, in case something nuts happens.